deeplearning4j-scaleout/spark/dl4j-spark-parameterserver/src/main/java/org/deeplearning4j/spark/parameterserver/networking/SilentTrainingDriver.java

```diff
@@ -139,7 +142,19 @@ public void startTraining(SilentUpdatesMessage message) {
             // master invokes everything, since that's Silent Worker approach: we want master to be always up-to-date
             synchronized (this) {
                 // threshold decoder is inplace & fast
-                Nd4j.getExecutioner().thresholdDecode(message.getUpdates(), updates);
+                int encoding = message.getUpdates().data().getInt(3);
+                if (encoding == 0) {
+                    Nd4j.getExecutioner().thresholdDecode(message.getUpdates(), updates);
+                    sparseCounter.incrementAndGet();
+                } else if (encoding == 1) {
+                    Nd4j.getExecutioner().bitmapDecode(message.getUpdates(), updates);
+                    denseCounter.incrementAndGet();
+                }
+
+                if ((sparseCounter.get() + denseCounter.get()) % 100 == 0) {
+                    log.info("Sparse/Dense ratio: {}", String.format("%.2f", (sparseCounter.get() +1) / (double) (denseCounter.get() + 1)));
```

Comment:
Debug level for the log?
In the future (UI pass) we might look at tracking this info over time...